src/App.js

```diff
@@ -27,6 +27,20 @@ function App() {
   const [uid, setUid] = useState(null);
   const [user, setUser] = useState(null);
   const [themeType, setThemeType] = useStorage("theme", "light");
+  const [customLightTheme, setCustomLightTheme] = useState(lightTheme);
+  const [customDarkTheme, setCustomDarkTheme] = useState(darkTheme);
+  const [customColors] = useStorage("customColors", null);
```

Comment:
Why are you defining customColors in both the Navbar and the App state? Isn't this a bit broken? I would imagine that you could just have a handler to change customColors in the App state, and the `useEffect` below this line would update the theme automatically whenever the colors change.